### PR TITLE
Add snippet explaining lazy-loading inside an MDX file.

### PR DESCRIPTION
**Pull request by joshwcomeau**
## Description

Yesterday evening I was chatting with @johno about lazy-loading content from within `.mdx` files, and came up with a workable solution (eventually gatsby-plugin-mdx will address this directly, but for now). Added a quick blurb + snippet about it to the docs.

Tried to use the Hemingway app to get it down to Grade 6 level, couldn't get below Grade 9 😅 feedback appreciated!

### CODE REVIEW

File: docs/docs/mdx/importing-and-using-components.md [hunk 1]
```diff
@@ -73,6 +73,48 @@ Because the `<Message />` and `<Chart />` components were passed into the provid
...
+}
+
+export default LazyThing
+```
+
+Inside your MDX, swap out any references to `Thing` with `LazyThing`:
+
+```diff
+-import Thing from "../components/Thing/Thing.js"
++import LazyThing from "../components/Thing/LazyThing.js"
```
Comment (johno) [inline docs/docs/mdx/importing-and-using-components.md]: I wonder if we should keep the component as `Thing` and only change the import? This has the advantage of not changing the component usage in the MDX file.

File: docs/docs/mdx/importing-and-using-components.md [hunk 2]
```diff
@@ -73,6 +73,48 @@ Because the `<Message />` and `<Chart />` components were passed into the provid
   lessonTitle="Make React components globally available as shortcodes in MDX"
 />
 
+## Lazy-loading components
+
+When you use components in your `.mdx` files, Gatsby will bundle them into the main application bundle. This can cause performance problems.
+
+In the future, [`gatsby-plugin-mdx`](/packages/gatsby-plugin-mdx) will address this. In the meantime, it can be prudent to lazy-load very large dependencies. The following snippet provides an example for lazy-loading an imaginary `Thing` component:
+
+```jsx:title=src/components/Thing/LazyThing.js
```
Comment (laurieontech) [inline docs/docs/mdx/importing-and-using-components.md]: Silly thing, but I feel like the path being this long might be unecessary? can it be `src/components/LazyThing.js` just to make it a bit more readable?

File: docs/docs/mdx/importing-and-using-components.md (hunk 1 above)
Comment (joshwcomeau) [reply docs/docs/mdx/importing-and-using-components.md]: Yeah, I was debating this myself! In fact this is what I did in my blog when I had this problem for real. My hesitation was that it makes this thing more implicit, which might be confusing for folks on a shared codebase, or someone who forgets that it's being lazy-loaded and forgets to lazy-load their next big component.

But yeah I don't have strong opinions! @laurieontech what do you think?

**Comment by joshwcomeau** [comment]
> I also wonder if it's fair to include use of a Spinner if that's not something we explain? What happens if you don't give it anything? Or an empty component of some kind?

@laurieontech good point! What if I make a `Placeholder` component, and define it near the top of the snippet?

```js
const Placeholder = () => null;
```

File: docs/docs/mdx/importing-and-using-components.md (hunk 1 above)
Comment (laurieontech) [reply docs/docs/mdx/importing-and-using-components.md]: If I'm writing it myself I probably keep it as Thing. But for the benefit of drawing attention to what is happening in the docs I'd likely go for the LazyThing option.

**Comment by laurieontech** [comment]
Ya @joshwcomeau that feels more accessible!

**Comment by laurieontech** [comment]
Let me know if you're still making changes @joshwcomeau. This is looking really good!

**Comment by joshwcomeau** [comment]
I think I'm done =) let me know if there's anything else I should change!

(also I'm guessing the failing tests are unrelated to my work but lemme know if there's anything I need to do on that front)

**Comment by laurieontech** [comment]
Thanks Josh!

**Pull request merged**